### PR TITLE
Add ETH flow order converter

### PR DESCRIPTION
**Pull request by fedgiac**
Adds a library that define the format of an ETH flow order and maps it to the CoW Swap order that will be settled onchain.

### Test plan

Unit tests.

See [specifications](https://docs.google.com/document/d/1D9P6A-X_sjZyV7i_f7XTZx5o7znFgRbNVJghHtBcy7U/edit#heading=h.fu7l92qg4u5x) at `User and handler orders` for comparison.

### CODE REVIEW

File: src/libraries/EthFlowOrder.sol [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    /// @param wrappedNativeToken The address of the wrapped native token for the current network (e.g., WETH for
+    /// Ethereum mainet).
+    /// @return The CoW Swap order data that represents the user order in the ETH flow contract.
+    function toCoWSwapOrder(Data calldata order, IERC20 wrappedNativeToken)
+        public
+        pure
+        returns (GPv2Order.Data memory)
+    {
+        if (order.receiver == GPv2Order.RECEIVER_SAME_AS_OWNER) {
+            // The receiver field specified which address is going to receive the proceeds from the oders. If using
```
Comment (josojo) [inline src/libraries/EthFlowOrder.sol:50]: nit: spelling orders.

File: src/libraries/EthFlowOrder.sol [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    /// @param order The ETH flow order to be converted.
+    /// @param wrappedNativeToken The address of the wrapped native token for the current network (e.g., WETH for
+    /// Ethereum mainet).
+    /// @return The CoW Swap order data that represents the user order in the ETH flow contract.
+    function toCoWSwapOrder(Data calldata order, IERC20 wrappedNativeToken)
+        public
+        pure
+        returns (GPv2Order.Data memory)
+    {
+        if (order.receiver == GPv2Order.RECEIVER_SAME_AS_OWNER) {
```
Comment (josojo) [inline src/libraries/EthFlowOrder.sol:49]: Should we also check that the order.receiver is != the null address?

File: test/FillWithSameByte.sol [hunk 3]
```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8;
+
+library FillWithSameByte {
```
Comment (josojo) [inline test/FillWithSameByte.sol:4]: I like that! I think its good practice. Maybe on eco-system Friday, we can make a PR to foundry to add it by default.

File: src/libraries/EthFlowOrder.sol (hunk 2 above)
Comment (josojo) [reply src/libraries/EthFlowOrder.sol:49]: Ah, I just saw:     
```
address internal constant RECEIVER_SAME_AS_OWNER = address(0);
```
All good.

**Review by josojo — APPROVED** [review]
Great! I like the gas optimization and security check

**Review by nlordell — APPROVED** [review]
Code looks good.

**Pull request merged**